nexusiq/src/main/kotlin/NexusIqService.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.nexusiq
+
+import com.fasterxml.jackson.annotation.JsonAlias
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import okhttp3.OkHttpClient
+
+import org.ossreviewtoolkit.model.Vulnerability
+import org.ossreviewtoolkit.utils.BasicAuthInterceptor
+
+import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+/**
+ * Interface for the NexusIQ REST API, based on the documentation from
+ * https://help.sonatype.com/iqserver/automating/rest-apis.
+ */
+interface NexusIqService {
+    companion object {
+        /**
+         * Create a NexusIQ service instance for communicating with a server running at the given [url],
+         * optionally using a pre-built OkHttp [client].
+         */
+        fun create(
+            url: String,
+            user: String? = null,
+            password: String? = null,
+            client: OkHttpClient? = null
+        ): NexusIqService {
+            val authenticatedClient = (client?.newBuilder() ?: OkHttpClient().newBuilder()).apply {
+                if (user != null && password != null) {
+                    addNetworkInterceptor(BasicAuthInterceptor(user, password))
+                }
+            }.build()
+
+            val retrofit = Retrofit.Builder()
+                .client(authenticatedClient)
+                .baseUrl(if (url.endsWith("/")) url else "$url/")
```

Comment:
Is appending the slash *really* necessary? I'd be very surprised if Retrofit didn't do this internally.

Reply:
Checked it again. You're right, it's not required.